android/MobiledgeXSDKDemo/matchingenginehelper/src/main/java/com/mobiledgex/matchingenginehelper/MatchingEngineHelper.java

```diff
@@ -492,11 +491,140 @@ public void registerClientInBackground() {
         }).start();
     }
 
+    public boolean qosPrioritySessionCreate(String qosProfile, int duration) throws ExecutionException, InterruptedException,
+            io.grpc.StatusRuntimeException, PackageManager.NameNotFoundException {
+        if (!verifyMeRequirements()) {
+            String message = "Matching engine requirements not met. Please allow app permissions " +
+                    "and enable Enhanced Location Verification in Matching Engine Settings";
+            Log.e(TAG, message);
+            meHelperInterface.showError(message);
+            return false;
+        }
+        AppClient.QosPrioritySessionCreateRequest.Builder builder = me.createDefaultQosPrioritySessionCreateRequest(mActivity);
+        builder.setPortApplicationServer(mTestPort+"");
+        builder.setIpUserEquipment(mDeviceIpv4);
+        builder.setIpApplicationServer(mAppInstIp);
+        builder.setProtocolIn(AppClient.QosSessionProtocol.valueOf(mAppInstProto));
+        builder.setProfile(AppClient.QosSessionProfile.valueOf(qosProfile));
+        builder.setSessionDuration(duration);
+
+        AppClient.QosPrioritySessionCreateRequest qosPrioritySessionCreateRequest;
+        Future<AppClient.QosPrioritySessionReply> qosPrioritySessionReplyFuture;
+        qosPrioritySessionCreateRequest = builder.build();
+        qosPrioritySessionReplyFuture = me.qosPrioritySessionCreateFuture(qosPrioritySessionCreateRequest, mDmeHostname, mDmePort,10000);
+        Log.i(TAG, "qosPrioritySessionReplyFuture="+qosPrioritySessionReplyFuture);
+        mQosPrioritySessionReply = qosPrioritySessionReplyFuture.get();

```

Comment:
This retrieves a future, then call .get(), defeating reason for Future along async paths. A completableFuture is more chain able?